### ISSUE TITLE
A question on learning rate decay schedule

### ISSUE THREAD

**Issue by zyushun** [jzhang38/TinyLlama#172]
For the learning rate decay schedule, why do you use  "lr = get_lr(state["iter_num"]) if decay_lr else learning_rate"?  Here, "state["iter_num"]" is the number of processed minibatch, not the number of training steps.  

By definition of decay learning rate, shouldn't we use  "state["step_count"]" instead of ""state["iter_num"]"?

I am talking about line 213 in [TinyLlama](https://github.com/jzhang38/TinyLlama/tree/main)/[pretrain](https://github.com/jzhang38/TinyLlama/tree/main/pretrain)/tinyllama.py

**Comment by jzhang38** [comment]
Yeah, we indeed use iter_num, which is the minibatch number to calculate the learning rate.

I believe it has minimal effect on the training loss.
